apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/logging/LoggingFilter.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.api.common.logging;
+
+import static io.stargate.sgv2.api.common.config.constants.LoggingConstants.*;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import io.stargate.sgv2.api.common.config.LoggingConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerResponseContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Collections;
+import java.util.Set;
+import org.jboss.resteasy.reactive.server.ServerResponseFilter;
+import org.slf4j.Logger;
+
+/** The filter for logging requests when required for debugging purposes. */
+@ApplicationScoped
+public class LoggingFilter {
+  private static final Logger logger = org.slf4j.LoggerFactory.getLogger(LoggingFilter.class);
+
+  /** The request info bean. */
+  private final StargateRequestInfo requestInfo;
+  /** The configuration for logging. */
+  private final LoggingConfig loggingConfig;
+
+  /** Default constructor. */
+  @Inject
+  public LoggingFilter(StargateRequestInfo requestInfo, LoggingConfig loggingConfig) {
```

Comment:
@kathirsvn Does this (injection of `StargateRequestInfo` mean that a new filter is created for every request)? I think it must mean that... is there no way around needing to get that injected? And instead get if from `ContainerRequestContext`?

Reply:
@tatu-at-datastax Great question! this discovers another issue I think.

`StargateRequestInfo` is `RequestScoped` and `LoggingFilter` is `ApplicationScoped`. So, if the `LoggingFilter` is designed to be shared across requests, I'm not sure how the request specific `StargateRequestInfo` ll be passed on cleanly.

I think we should either change `LoggingFilter` to `RequestScoped` or we should remove support for tenant filter in this feature, because parsing tenant from `ContainerRequestContext` or `ContainerResponseContext` is not possible I believe. I could see we use `RoutingContext` for resolving tenant info in `TenantResolver`

Reply:
Also, some of the Filter optimization can be done if `LoggingFilter` is going to be created only once in the applications life cycle.

Reply:
I will do some more debugging on what information we can get from `ContainerRequestContext` and `ContainerResponseContext` and see if getting tenant id be still achieved while leaving `LoggingFilter` as `ApplicationScoped`

Reply:
On mis-matching scopes: Quarkus use proxies to handle this case, as I recall -- so marking it as "Application-scoped" does not fail, but requires use of proxy to delegate from single instance to per-request implementations. Or such.

But I think that since Tenant Id is not easily accessible (maybe it's injected by another filter? I am not sure how that is done), we probably just need to accept per-request construction.
LMK what you find!

Reply:
@tatu-at-datastax 
I agree on the proxies part. I just debugged this and I can confirm that using proxies, the request scoped beans are updated on the ApplicationScoped instances.

Though the constructor of `LoggingFilter` is invoked only once on the first request, the `StargateRequestInfo` keeps changing when the `@ServerResponseFilter` method is invoked. i.e. The request scoped bean is changing as expected for each request. I changed the token between requests and I could see it changing when it reaches the filter method on that instance, so we are good, no need to change the filter to `RequestScoped`

Also, there is a predecessor for this design. `TenantRequestMetricsFilter` (which is `ApplicationScoped`) works the same way. So, I'm merging this.